lib/service/db.go

```diff
@@ -111,7 +111,7 @@ func (process *TeleportProcess) initDatabaseService() (retErr error) {
 
 	clusterName := conn.ServerIdentity.Cert.Extensions[utils.CertExtensionAuthority]
 
-	authorizer, err := auth.NewAuthorizer(clusterName, conn.Client, conn.Client, conn.Client)
+	authorizer, err := auth.NewAuthorizer(clusterName, accessPoint)
```

Comment:
So I'm thinking from the perspective of the database agent specifically but this may apply to other agents as well.

This authorizer is used infrequently AFAIU -  only when establishing a new database session - so does it really make sense to use the cache here? We'd be hardly reducing any load but potentially introducing subtle authz issues related to cache staleness, etc.

Shouldn't all authorization logic use the database directly as the source of truth to be on the safe side? Is this really what's causing all the load on dynamodb?

Reply:
The cache is always used because even requests to auth don't hit the backend directly, they hit auth's cache first. So if we have bugs in the cache, they'll be hit regardless.

Good point about database access, but should help Application and Kubernetes Access. For Application Access I beleve RBAC is check on every request and for Kubernetes every time you do `kubectl exec` RBAC is checked.

Reply:
Yeah makes sense. Now that I'm thinking more about this, it should actually help even with database access. We're already checking each Mongo query against RBAC, and potentially will extend this to check queries for other protocols as well in future.